### PR TITLE
Beats: Move identical method implementations from BeatGrid/BeatMap

### PR DESCRIPTION
**Pull request by Holzhaus**
This is a first step toward merging the beatgrid/beatmap classes into a
single class.

### CODE REVIEW

File: src/track/beats.h [hunk 1]
```diff
@@ -155,6 +155,10 @@ class Beats {
 
     /// Adjust the beats so the global average BPM matches `bpm`.
     virtual BeatsPointer setBpm(mixxx::Bpm bpm) = 0;
+
+  protected:
+    /// For internal use only.
```
Comment (Be-ing) [inline src/track/beats.h:160]: This comment can be removed. That is clear from the method being protected.

File: src/track/beatgrid.cpp [hunk 2]
```diff
@@ -143,41 +143,6 @@ bool BeatGrid::isValid() const {
     return m_sampleRate.isValid() && bpm().isValid() && firstBeatPosition().isValid();
 }
 
-// This could be implemented in the Beats Class itself.
-// If necessary, the child class can redefine it.
```
Comment (Be-ing) [inline src/track/beatgrid.cpp:147]: Why write this comment instead of just doing it...

Comment (uklotzde) [reply src/track/beatgrid.cpp:147]: @Be-ing I guess you got confused by the diff? This code and comments have been deleted because they became obsolete.

Comment (Be-ing) [reply src/track/beatgrid.cpp:147]: I saw that. I am just lamenting that this wasn't already done 10 years ago.

File: src/track/beatmap.h [hunk 3]
```diff
@@ -82,7 +79,7 @@ class BeatMap final : public Beats {
     BeatMap(const BeatMap& other);
 
     // For internal use only.
```
Comment (uklotzde) [inline src/track/beatmap.h:81]: Same here, adding such a comment to a private method is confusing. Unrelated, but let's clean this up now.

**Review by uklotzde — APPROVED** [review]
Ready after deleting the remaining, useless comments.

**Comment by Be-ing** [comment]
thanks

**Pull request merged**